scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -806,176 +951,27 @@
-  (let [view-offers-atom (atom [])
-        view-offers-fn (fn [] @view-offers-atom)]
-    (async/go
-     (try
-       (loop [offers []]
-         (reset! view-offers-atom
-                 offers)
-         (recur
-          (if (seq offers)
-            (do
-              (log/debug (count offers) " offers in prioritizer buffer")
-              (async/alt!
-               offer-ready-chan ([offer]
-                                   (log/debug "Got new offer. Size of prioritizer buffer: "
-                                              (inc (count offers)))
-                                   (counters/inc! prioritizer-buffer-size)
-                                   (sort-offers (conj offers offer)))
-               [[matcher-chan (first offers)]] (do
-                                                 (log/debug "Offer sent to matcher. "
-                                                            (count (rest offers))
-                                                            " offers remaining")
-                                                 (counters/dec! prioritizer-buffer-size)
-                                                 (rest offers))))
-            (let [offer (async/<! offer-ready-chan)]
-              (log/debug "Got new offer. Size of prioritizer buffer: "
-                         (inc (count offers)))
-              (counters/inc! prioritizer-buffer-size)
-              (sort-offers (conj offers offer))))))
-       (catch Exception e
-         (log/error e "In start-offer-prioritizer"))))
-    view-offers-fn))
-
-(defn start-offer-matcher!
-  "Create a thread to repeatedly pull from matcher-chan and have the offer matched."
-  [conn pending-jobs-atom matcher-chan rescinded-offer-id-cache]
-  (async/thread
-    (log/debug "offer matcher started")
-    (try
-      (loop []
-        (log/debug "Preparing to pull offer (offer-matcher)")
-        (let [offer (async/<!! matcher-chan)
-              fid (:fid offer)
-              offer-ids (:ids offer)
-              driver (:driver offer)]
-          (log/debug "(offer-matcher) Offer taken " offer)
-          (if-let [invalid-offer-id (some #(cache/lookup @rescinded-offer-id-cache %) offer-ids)]
-            (log/info "Received rescinded offer" invalid-offer-id "in offer matcher")
-            (handle-resource-offer! conn driver fid pending-jobs-atom offer)))
-        (log/debug "(offer matcher) Handled resource offer")
-        (recur))
-      (catch Exception e
-        (log/error e "Error in start-offer-matcher!")))))
-
-(meters/defmeter [cook-mesos scheduler offer-back-pressure])
 (meters/defmeter [cook-mesos scheduler mesos-error])
 
+(defn make-fenzo-scheduler
+  [driver offer-incubate-time-ms]
+  (.. (com.netflix.fenzo.TaskScheduler$Builder.)
+      (disableShortfallEvaluation) ;; We're not using the autoscaling features
+      (withLeaseOfferExpirySecs (max (quot offer-incubate-time-ms 1000) 1)) ;; Must convert from the ms, should be at least 1 second
```

Comment:
Use `(-> offer-incubate-time-ms t/millis t/in-seconds)` instead of `(quot offer-incubate-time-ms 1000)`